main/background.js

```diff
@@ -52,146 +55,102 @@ async function showDialog(title, message) {
-        ],
-        max_tokens: maxTokens ? maxTokens : 500,
-        temperature: temperature ? temperature : 0.75,
-        stream: true
-      }
-    };
-    console.log(options);
-    let bufferData = '';
-    function get_content_block(block) {
-      try {
-          const jsonData = JSON.parse('{ "data"' + block.trim().slice(4, -1) + "}}");
-          bufferData = jsonData["data"]["choices"][0]["message"]["content"];
-          return bufferData;
-      } catch (error) {
-          // console.log("Error happened: Skipped and will catch up with the next block.");
-      }
-      console.log("||||||||||||||||||||||||||||||||||||||||||||||", bufferData);
-      return bufferData;
+ipcMain.on("search", async (event, query, model, token, systemPrompt, temperature, maxTokens) => {
+  const options = {
+    method: 'POST',
+    url: 'https://api.perplexity.ai/chat/completions',
+    headers: {
+      accept: 'application/json',
+      'content-type': 'application/json',
+      authorization: `Bearer ${token}`
+    },
+    responseType: 'stream',
+    data: {
+      model: model,
+      messages: [
+        {role: 'system', content: systemPrompt || 'Be precise and concise.'},
+        {role: 'user', content: query}
+      ],
+      max_tokens: maxTokens ? parseInt(maxTokens, 10) : 500,
+      temperature: temperature ? parseFloat(temperature) : 0.75,
+      stream: true
     }
-    
-    try {
-      const response = await axios.request(options)
-      const stream = response.data;
-      let buffer = '';
-  
-      stream.on('data', (chunk) => {
-        // Convert the chunk to a string and remove leading/trailing/whitespace
-        let read_data = chunk.toString('utf-8'); 
-        let last_8_char = read_data.slice(-8).trim();
-        let top_12_char = read_data.slice(0, 12).trim();
-
-        // itt might contains more than 0 blocks
-        let blocks = get_blocks(read_data); 
-        let block_size = blocks.length;
-
-        if (last_8_char === "}}]}" && block_size === 1 && top_12_char === 'data: {"id":') {
-          const answer = get_content_block(blocks[0]);
-          event.sender.send('search-result', answer);
-          buffer = ""
-
-        } else { // this else stands for the case when the data is not complete
-
-          if (block_size >= 1) { 
-            const answer = get_content_block(blocks[blocks.length - 1]);
-            event.sender.send('search-result', answer);
-          }
-          buffer += read_data; // no need anymore
-        }
-      });
-
-      stream.on('end', () => {
-          event.sender.send('search-end');
-      });
+  };
+
+  try {
+    const response = await axios.request(options);
+    const stream = response.data;
+    let buffer = "";
+
+    stream.on('data', (chunk) => {
+      buffer += chunk.toString('utf-8');
+      let blocks = get_blocks(buffer);
+      if (blocks.length > 0) {
+        const answer = get_content_block(blocks[blocks.length - 1]);
```

Comment:
This works well if we parse only text answers. But mixed text and code answers don't render very well in the UI and sometimes, it contains incomplete answers.

Reply:
it works, fine. my bad, i keep your version :)